demos/python_demos/mri_segmentation.py

```diff
@@ -0,0 +1,303 @@
+from the Medical Decathlon dataset. The Medical
+Decathlon dataset (http://medicaldecathlon.com/) is
+distributed under the Creative Commons Attribution-ShareAlike 
+International 4.0 license
+(https://creativecommons.org/licenses/by-sa/4.0/).
+
+You'll need the extension library to handle the Resize_Bilinear operations.
+
+python inference_openvino.py -l ${INTEL_CVSDK_DIR}/inference_engine/lib/centos_7.4/intel64/libcpu_extension_avx2.so
+
+"""
+
+import sys
+import os
+from argparse import ArgumentParser
+import numpy as np
+import logging as log
+from time import time
+from openvino.inference_engine import IENetwork, IEPlugin
+
+def dice_score(pred, truth):
+    """
+    Sorensen Dice score
+    Measure of the overlap between the prediction and ground truth masks
+    """
+    numerator = np.sum(pred * truth) * 2.0 + 1.0
+    denominator = np.sum(pred) + np.sum(truth) + 1.0
+
+    return numerator / denominator
+
+def plot_predictions(predictions, input_data, label_data, img_indicies, args):
+    """
+    Plot the predictions with matplotlib and save to png files
+    """
+    png_directory = "inference_examples_openvino"
+    if not os.path.exists(png_directory):
+        os.makedirs(png_directory)
+
+    import matplotlib.pyplot as plt
+
+    # Processing output blob
+    log.info("Plotting the predictions and saving to png files. Please wait...")
+    number_imgs = predictions.shape[0]
+    num_rows_per_image = args.rows_per_image
+    row = 0
+
+    for idx in range(number_imgs):
+
+        if row==0:  plt.figure(figsize=(15,15))
+
+        plt.subplot(num_rows_per_image, 3, 1+row*3)
+        plt.imshow(input_data[idx,0,:,:], cmap="bone", origin="lower")
+        plt.axis("off")
+        if row==0: plt.title("MRI")
+
+        plt.subplot(num_rows_per_image, 3, 2+row*3)
+        plt.imshow(label_data[idx,0,:,:], origin="lower")
+        plt.axis("off")
+        if row==0: plt.title("Ground truth")
+
+        plt.subplot(num_rows_per_image, 3, 3+row*3)
+        plt.imshow(predictions[idx,0,:,:], origin="lower")
+        plt.axis("off")
+        if row ==0:  plt.title("Prediction")
+
+        plt.tight_layout()
+
+        if (row == (num_rows_per_image-1)) or (idx == (number_imgs-1)):
+
+            if num_rows_per_image==1:
+                fileidx = "pred{}.png".format(img_indicies[idx])
+            else:
+                fileidx = "pred_group{}".format(idx // num_rows_per_image)
+            filename = os.path.join(png_directory, fileidx)
+            plt.savefig(filename,
+                        bbox_inches="tight", pad_inches=0)
+            print("Saved file: {}".format(filename))
+            row = 0
+        else:
+            row += 1
+
+def load_data():
+    """
+    Modify this to load your data and labels
+    """
+
+    # Load data
+    # You can create this Numpy datafile by running the create_validation_sample.py script
+    data_file = np.load("mri_segmentation_data/validation_data.npz")
```

Comment:
That would require the additional steps of the user downloading the entire BraTS dataset (several GB) and then pre-processing the data just to pull out a few dozen 2D slices. Instead, it makes more sense to save those few dozen slices in a NumPy array (which is shipped in the model zoo like we do for car.bmp in the license plate identification model). I provide the Nifti processing scripts in the training repo (https://github.com/IntelAI/unet) for those that want to re-train the model or pull out other data, but as this is just a model zoo to show a few sample cases, it seemed like overkill to download several GB of data for 12 slices.

Reply:
That wouldn't require additional steps since instead of the single numpy file we can ship single nifti series. This will make the whole pipeline (load data -> preprocess -> infer -> postprocess -> save) more clear.

Reply:
@tonyreina I think, we can go with your logic about making the sample more friendly to "casual" users. So, plotting and output can, probably, stay as is. For input, though, I'd agree with @lachinov - it makes more sense to take "images" in the original format (i.e. make your conversion logic a part of this script). So, instead of storing a single ("sample") `.npz` you'd store a single series of `.nifti`s. This way it doesn't introduce any additional overhead (i.e. downloading of several GBs). Makes sense?

Btw, as for storing the Brats data itself - we, probably, need to take care about that. Both in terms of binaries in git and potential license issues. Thinking of options - can you include 1 example series (same 10Mb you have in the PR, but in nifti format) to the repo you're planning to release?